indigo/indigo/src/main/scala/indigo/shared/input/Pointers.scala

```diff
@@ -0,0 +1,24 @@
+package indigo.shared.input
+
+import indigo.shared.collections.Batch
+import indigo.shared.datatypes.Point
+import indigo.shared.datatypes.Rectangle
+import indigo.shared.events.PointerEvent
+
+final class Pointers(
+    val pointerEvents: Batch[PointerEvent],
+    val position: Point
+)
+
+object Pointers:
+  val default: Pointers =
+    Pointers(Batch.empty, Point.zero)
+
+  def calculateNext(previous: Pointers, events: Batch[PointerEvent]) =
+    Pointers(
+      pointerEvents = events,
+      position = lastPointerPosition(previous.position, events)
+    )
+
+  private def lastPointerPosition(previous: Point, events: Batch[PointerEvent]): Point =
+    events.collect { case mp: PointerEvent.PointerMove => mp.position }.lastOption.fold(previous)(identity)
```

Comment:
No need to change it, just a comment: I've seen fold used like this a few times recently and I just don't understand how it's syntactically desirable over `getOrElse`. Maybe you can enlighten me on it's appeal?

Reply:
It was just a straight forward pattern match at first which I directly translated to fold, but yeah the `getOrElse` could have been even better.  I can change it later.